### DISCUSSION TITLE
Colouring headers: allow regex with context

### DISCUSSION THREAD

**Discussion by whitney-cumber** [neomutt/neomutt#4523] [Development]
## What would you like to be able to do?

I'd like to colour the header field name in a different colour than the header field value, e.g. for "Subject: foo" I'd like to colour "Subject:" blue and "foo" red, while doing this with different colours depending on the header field.  Example:

```
Blue-From: Red email address
Blue-To: Green email address
White-Subject: Orange subject text
blue-other-stuff: Grey other value
blue-other-stuff-2: Grey other value 2
```

## What needs to change in NeoMutt?

Some kind of context pattern matches similar to how the status line supports it.

## Additional Info

Please without the caveats mentioned in: https://github.com/neomutt/neomutt/issues/3672

Similar to:
https://github.com/neomutt/neomutt/issues/3660 (allow different colours in attachment menu)
https://github.com/neomutt/neomutt/issues/3617 (allow different colours in sidebar)

Edit: Clarified feature request

**Comment by jindraj** [comment]
this already exists, see [$header_color_partial](https://neomutt.org/guide/configuration#11-3-%C2%A0color-lists)

example:
```
set   header_color_partial = yes
color hdrdefault  brightcyan     default
color header      yellow         default  "^[[:alpha:]][^:]+:"
color header      brightyellow   default  "^(From|To|CC|Bcc|Subject):"
```

This is how it looks like
<img src="https://i.ibb.co/VD8bDkt/Screenshot-2023-02-09-at-10-10-33.png" alt="Screenshot-2023-02-09-at-10-10-33" border="0">

**Comment by auouymous** [comment]
That technically does exactly what @whitney-cumber described above, but it prevents coloring any values different from the others. Setting a color for `^Subject: .*` with `header_color_partial = yes` highlights the entire first line, and leaves all other lines of subject as the hdrdefault color. And a second rule before or after won't color just "Subject:" to a different color.

**Comment by whitney-cumber** [comment]
I guess I was to vague when describing [I updated the original description]. I want to colour the *value* of *two* different subjects with **two** different colours and the field *name* in another colour. AFAIK that is currently not possible (though happy to be proven wrong).

Example:

```
Blue-From: Red email address
Blue-To: Green email address
White-Subject: Orange subject text
blue-other-stuff: Grey other value
blue-other-stuff-2: Grey other value 2
```
